src/overlays/gamestates/ovl_title/z_title.c

```diff
@@ -1,22 +1,183 @@
 /*
  * File: z_title.c
  * Overlay: ovl_title
- * Description:
+ * Description: Nintendo 64 logo shown on startup
  */
 
 #include "z_title.h"
+#include "overlays/gamestates/ovl_opening/z_opening.h"
+#include "static/nintendo_rogo_static/nintendo_rogo_static.h"
 
-extern UNK_TYPE D_01002720;
+void Title_UpdateCounters(TitleContext *this) {
+    if ((this->coverAlpha == 0) && (this->visibleDuration != 0)) {
+        this->timer--;
+        this->visibleDuration--;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_title/Title_UpdateCounters.s")
+        if (this->timer == 0) {
+            this->timer = 400;
+        }
+    } else {
+        this->coverAlpha += this->addAlpha;
+        
+        if (this->coverAlpha <= 0) {
+            this->coverAlpha = 0;
+            this->addAlpha = 12;
+        } else if (this->coverAlpha >= 255) {
+            this->coverAlpha = 255;
+            this->exit = true;
+        }
+    }
+    
+    this->uls = this->ult & 0x7F;
+    this->ult++;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_title/Title_RenderView.s")
+void Title_RenderView(TitleContext *this, f32 x, f32 y, f32 z) {
+    View* view = &this->view;
+    Vec3f eye;
+    Vec3f lookAt;
+    Vec3f up;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_title/Title_Draw.s")
+    eye.x = x;
+    eye.y = y;
+    eye.z = z;
+    up.x = up.z = 0.0f;
+    lookAt.x = lookAt.y = lookAt.z = 0.0f;
+    up.y = 1.0f;
+    func_8013F0D0(view, 30.0f, 10.0f, 12800.0f);
+    View_SetViewOrientation(view, &eye, &lookAt, &up);
+    View_RenderView(view, 0xF);
+}
 
-void Title_Main(GameState* thisx);
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_title/Title_Main.s")
+void Title_Draw(GameState *thisx) {
+    static s16 titleRotation = 0;
+    static Lights1 sTitleLights = gdSPDefLights1(100, 100, 100, 255, 255, 255, 69, 69, 69);
+    TitleContext* this = (TitleContext*)thisx;
 
```

Comment:
```suggestion
    TitleContext* this = (TitleContext*)thisx;
```

Reply:
Not sure if it's showing up on github, but I don't think you need a space between variable declarations